readthedocs/projects/tasks/search.py

```diff
@@ -40,112 +42,215 @@ def fileify(version_pk, commit, build, search_ranking, search_ignore):
         version_slug=version.slug,
     )
     try:
-        _create_imported_files(
+        _create_imported_files_and_search_index(
             version=version,
-            commit=commit,
-            build=build,
+            build_id=build,
             search_ranking=search_ranking,
             search_ignore=search_ignore,
         )
     except Exception:
         log.exception('Failed during ImportedFile creation')
 
+
+@app.task(queue="reindex")
+def index_build(build_id):
+    """Create imported files and search index for the build."""
+    build = (
+        Build.objects.filter(pk=build_id)
+        .select_related("version", "version__project")
+        .first()
+    )
+    if not build:
+        return
+
+    # The version may have been deleted.
+    version = build.version
+    if not version:
+        return
+
+    log.bind(
+        project_slug=version.project.slug,
+        version_slug=version.slug,
+        build_id=build.id,
+    )
+
+    search_ranking = []
+    search_ignore = []
+    build_config = build.config
+    if build_config:
+        search_config = build_config.get("search", {})
+        search_ranking = search_config.get("ranking", [])
+        search_ignore = search_config.get("ignore", [])
```

Comment:
Small simplification:

```suggestion
    build_config = build.config or {}
    search_config = build_config.get("search", {})
    search_ranking = search_config.get("ranking", [])
    search_ignore = search_config.get("ignore", [])
```